jobs/openshift4/cluster/deprovision/Jenkinsfile

```diff
@@ -0,0 +1,98 @@
+                    git branch: "master", url: "https://gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins.git"
+                    clusterConfigExists = fileExists(clusterInfoDirectory)
+                }
+                if (clusterConfigExists) {
+                    rootWorkspaceDir = sh (
+                        script: "pwd",
+                        returnStdout: true
+                    ).trim()
+                    installConfigDirectory = rootWorkspaceDir + "/rhmi-jenkins/" + clusterInfoDirectory + "/"
+                    sh "openshift-install destroy cluster --dir=${installConfigDirectory} --log-level=${LOG_LEVEL} 2>&1 | tee log.file"
+                    String clusterDestroyOutput = readFile("log.file")
+                    if (clusterDestroyOutput.contains("FATAL")) {
+                        currentBuild.result = 'FAILURE'
+                    } else {
+                        withCredentials([string(credentialsId: "gitlab-access-token", variable: "GITLAB_ACCESS_TOKEN")]) {
+                            dir("rhmi-jenkins") {
+                                sh(
+                                    returnStdout: false,
+                                    script: """
+                                        rm -rf ${clusterInfoDirectory}
+                                        git remote add jenkins https://oauth2:${GITLAB_ACCESS_TOKEN}@gitlab.cee.redhat.com/integreatly-qe/rhmi-jenkins.git
+                                        git add -A
+                                        git commit -m 'cluster destroyed: ${CLUSTER_NAME}'
+                                        git push jenkins master
+                                    """
+                                )
+                            }
+                        }
+                    }
+                } else {
+                    error "No cluster terraform config files available for cluster with name: ${CLUSTER_NAME}!"
+                }
+            } // stage
+        } catch (caughtError){
+            currentBuild.result = "FAILURE"
+            err = caughtError
+            println caughtError
+        } finally {
+            notifyBuild(currentBuild.result, err)
+            cleanup()
+        }
+    }}} // timeout, ansiColor, timestamps
+} // node
+
+def notifyBuild(String buildStatus, err) {
+    if (params.RECIPIENTS.toString().length() > 0) {
+        buildStatus =  buildStatus ?: "SUCCESS"
+    
+        String mailRecipients = "${RECIPIENTS}"
+        String subject = "${buildStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'"
+        String body = "${subject}\n\n(${env.BUILD_URL})"
+
+        if (err != null) {
+            body = "${body}\n\n FAILURE was ${err}"
+        }
+
+        mail body: body, subject: subject, to: mailRecipients
+    }
+}
+
+def cleanup() {
+    cleanWs() // remove any files related to current build
+    sh (returnStdout: false, script: "rm -rf ~/.aws") // remove aws config folder
+}
```

Comment:
Quite a lot of duplication here with the cluster create pipeline. Could potentially move parts of this out into the library at some stage.